### DISCUSSION TITLE
Choosing Basis Sets in FLOSIC

### DISCUSSION THREAD

**Discussion by nerffan1** [FLOSIC/PublicRelease_2020#34] [Q&A]
Hello,

Is it possible to change basis sets in FLOSIC?  I see that the **NRLMOL_INPUT.DAT** file has the BASISV parameter, but I am not sure if the feature works. 

Thank You!

**Comment by yohyamamoto** [comment]
I checked this now and it is working but with a few caveats.

First, you need to define a linux environment value. This can go into your .bashrc or .zshrc.
`export NRLMOL_BASIS=<path to flosic code>/PublicRelease_2020/basis`

Then you can specify the basis set in the NRLMOL_INPUT.DAT
e.g.  `BASISV        = '6-311Gss'`

Basis sets supported out of the box are listed in the **getbasname.f90** subroutine. If it's not listed there, the filename gets ignored. Perhaps, we should change this to grab the string instead. 

At this point, starting a calculation from CLUSTER, you will see something like below in the screen out, and the ISYMGEN file reflects the basis set in action. 

```
 USING BASIS FILE:<path to the flosic code>/basis/6-311Gss.basis
 ATOM           25  NOT FOUND IN BASIS
 USING FROM DEFAULT BASIS SET
ATOM TYPE   1 (ALL-MAN): NUCLEAR CHARGE=  25,   1 ATOM(S)
 USING BASIS FILE:<path to the flosic code>/basis/6-311Gss.basis
ATOM TYPE   2 (ALL-OXY): NUCLEAR CHARGE=   8,   1 ATOM(S)
```
 
In this example, the basis set for O is found in the 6-311Gss.basis but not for Mn.

**Reply by yohyamamoto** [reply]
I just pushed a small change to the repository such that the subroutine uses basis set file name string internally. So you are not restricted by the list in the getbasname.f90. i.e. you can save your basis set in any file name and use it.

**Comment by jacks1ka** [comment]
Thanks, Yoh!

Sent from my iPhone

On Nov 14, 2023, at 6:36 PM, Yoh Yamamoto ***@***.***> wrote:

﻿

I checked this now and it is working but with a few caveats.

First, you need to define a linux environment value. This can go into your .bashrc or .zshrc.
export NRLMOL_BASIS=<path to flosic code>/PublicRelease_2020/basis

Then you can specify the basis set in the NRLMOL_INPUT.DAT
e.g. BASISV = '6-311Gss'

Basis sets supported out of the box are listed in the getbasname.f90 subroutine. If it's not listed there, the filename gets ignored. Perhaps, we should change this to grab the string instead.

At this point, starting a calculation from CLUSTER, you will see something like below in the screen out, and the ISYMGEN file reflects the basis set in action.

 USING BASIS FILE:<path to the flosic code>/basis/6-311Gss.basis
 ATOM           25  NOT FOUND IN BASIS
 USING FROM DEFAULT BASIS SET
ATOM TYPE   1 (ALL-MAN): NUCLEAR CHARGE=  25,   1 ATOM(S)
 USING BASIS FILE:<path to the flosic code>/basis/6-311Gss.basis
ATOM TYPE   2 (ALL-OXY): NUCLEAR CHARGE=   8,   1 ATOM(S)


In this example, the basis set for O is found in the 6-311Gss.basis but not for Mn.
